### ISSUE TITLE
A problem was reported in the nzbToRadarr.py script

### ISSUE THREAD

**Issue by weedaj** [clinton-hall/nzbToMedia#1546]
hi, I receive an error when pp-ing a failed download from nzbget to radarr. There are no further details shown even though debug is enabled. Below log is complete, no messages are logged after. It seems that Radarr does receive the failed download notification because alternative nzb's are pushed towards nzbget. Any idea what could be causing this?

```
2019-02-06 21:33:20 INFO    ::MAIN: Checking database structure...
2019-02-06 21:33:20 DEBUG   ::MAIN: Checking Initial Schema database upgrade
2019-02-06 21:33:20 DEBUG   ::MAIN: InitialSchema upgrade not required
2019-02-06 21:33:20 DEBUG   ::MAIN: Checking if we can use git commands: git version
2019-02-06 21:33:20 DEBUG   ::MAIN: Executing git version with your shell in /volume1/@appstore/nzbget/share/nzbget/scripts/nzbToMedia
2019-02-06 21:33:21 DEBUG   ::MAIN: git version : returned successful
2019-02-06 21:33:21 DEBUG   ::MAIN: Using: git
2019-02-06 21:33:21 DEBUG   ::MAIN: Executing git symbolic-ref -q HEAD with your shell in /volume1/@appstore/nzbget/share/nzbget/scripts/nzbToMedia
2019-02-06 21:33:21 DEBUG   ::MAIN: git symbolic-ref -q HEAD : returned successful
2019-02-06 21:33:21 INFO    ::MAIN: Checking if git needs an update
2019-02-06 21:33:21 DEBUG   ::MAIN: Executing git rev-parse HEAD with your shell in /volume1/@appstore/nzbget/share/nzbget/scripts/nzbToMedia
2019-02-06 21:33:22 DEBUG   ::MAIN: git rev-parse HEAD : returned successful
2019-02-06 21:33:22 DEBUG   ::MAIN: Executing git fetch origin with your shell in /volume1/@appstore/nzbget/share/nzbget/scripts/nzbToMedia
2019-02-06 21:33:30 ERROR   ::MAIN: Unable to contact github, can't check for update
2019-02-06 21:33:30 INFO    ::MAIN: No update needed
2019-02-06 21:33:31 INFO    ::MAIN: nzbToMedia Version:4896848099298ee2914760c56a66d4ad0674d181 Branch:master (Linux 2.6.32.12)
2019-02-06 21:33:32 WARNING ::MAIN: Failed to locate par2. Repair and rename using par files will not be possible!
2019-02-06 21:33:33 INFO    ::MAIN: #########################################################
2019-02-06 21:33:33 INFO    ::MAIN: ## ..::[nzbToMedia.py]::.. ##
2019-02-06 21:33:33 INFO    ::MAIN: #########################################################
2019-02-06 21:33:33 DEBUG   ::MAIN: Options passed into nzbToMedia: ['/volume1/@appstore/nzbget/share/nzbget/scripts/nzbToMedia/nzbToRadarr.py']
2019-02-06 21:33:33 INFO    ::MAIN: Script triggered from NZBGet Version 16.4.
2019-02-06 21:33:33 INFO    ::MAIN: Download failed with status FAILURE/HEALTH.
2019-02-06 21:33:33 DEBUG   ::MAIN: Adding NZB download info for directory /volume1/downloads/nzbget/inter/The.Wave.2015.720p.BluRay.x264-PSYCHD.#9035 to database
2019-02-06 21:33:34 INFO    ::MAIN: Auto-detected SECTION:Radarr
2019-02-06 21:33:34 INFO    ::MAIN: Calling Radarr:film to post-process:The.Wave.2015.720p.BluRay.x264-PSYCHD
2019-02-06 21:33:34 INFO    ::MAIN: Attemping imdbID lookup for The.Wave.2015.720p.BluRay.x264-PSYCHD
2019-02-06 21:33:34 INFO    ::MAIN: Searching folder and file names for imdbID ...
2019-02-06 21:33:34 INFO    ::MAIN: Searching IMDB for imdbID ...
2019-02-06 21:33:39 DEBUG   ::MAIN: Opening URL: http://www.omdbapi.com
2019-02-06 21:33:40 INFO    ::MAIN: Found imdbID [tt3616916]
2019-02-06 21:33:40 DEBUG   ::SERVER: Attempting to connect to server at http://localhost:8310/radarr/api/command
2019-02-06 21:33:40 DEBUG   ::SERVER: Server responded at http://localhost:8310/radarr/api/command
2019-02-06 21:33:40 WARNING ::RADARR: No media files found in directory /volume1/downloads/nzbget/inter/The.Wave.2015.720p.BluRay.x264-PSYCHD.#9035. Processing this as a failed download
2019-02-06 21:33:40 POSTPROCESS::RADARR: FAILED DOWNLOAD DETECTED FOR The.Wave.2015.720p.BluRay.x264-PSYCHD
2019-02-06 21:33:40 POSTPROCESS::RADARR: FAILED: The download failed. Sending failed download to Radarr for CDH processing
2019-02-06 21:33:40 ERROR   ::MAIN: A problem was reported in the /volume1/@appstore/nzbget/share/nzbget/scripts/nzbToMedia/nzbToRadarr.py script.
```

**Comment by clinton-hall** [comment]
is this a new install or has this just started happening on an existing install?

I found an issue recently where the transcoder was causing good videos to be marked as failed. this was fixed in latest nightly (although your logs don't appear to indicate that the transcoder was triggered.

otherwise, it appears this is not detecting any video files..
check autoProcessMedia.cfg to ensure that
```
[Extensions]
    mediaExtensions = .mkv,.avi,.divx,.xvid,.mov,.wmv,.mp4,.mpg,.mpeg,.vob,.iso,.m4v,.ts
```
includes the extensions for the types of videos you are downloading...

also, ensure nzbGet is set to repair and extract downloads, so that the video files are present before nzbToMedia is called.

**Comment by weedaj** [comment]
ty for your prompt reply. I guess it is a new install as I just switched from cp to radarr. Nzbget and nzbtomedia was already  existing but not used  in combination with radarr. I'm familiar with your postprocessing scripts (big fan) and everything seems to be configured well (including your points of attention).

nzbtomedia/nzbget either rightfully marked the movie marked as completed in radarr wo pp error or marked the movie as bad in radarr invoking new snatch but with pp error. 

In other words all seems to be working well (or I haven't figured out the impact yet :) ) but the error suggests otherwise.

**Comment by clinton-hall** [comment]
Ah... I see...

Yes, the difference is that for CP if/when a download fails nzbToMedia will tell CP it has failed and then exit with a success status.

For Radarr, because we can't tell radar that the download failed (or at least there was no method for this in the past and I haven't been told of any changes) the script essentially stops to let Radarr scan the empty directory as part of the Completed Download Handling (CDH).
The reason the script exists with an "error" is to flag it to you attention to check that it has been correctly processed. If this exists with success, the status shows in NZBGet as successful with nothing to alert you of the failure.

**Comment by weedaj** [comment]
ok, so works as designed? perhaps the exception message could be a bit more clear to prevent future questions but thanks for clarifying!

**Comment by clinton-hall** [comment]
I agree the message could be clearer, I'll take a look at this. To be clear, we force the "failed" status to ensure that Radarr doesn't try and process this as complete (not that it should, but for example a video file that is corrupted can be tricky, but nzbToMedia parses it thought ffprobe).

As far as I can see all has worked, but check at your end. After the script finished, did Radarr identify the download as failed and then snatch a new versions?

**Comment by weedaj** [comment]
yes, radarr identified it as failed judging from the fact that a new candidate was snatched. thanks for considering changing the notification and your support. Cheers, mate!

**Comment by clinton-hall** [comment]
With this logging change it should finish as follows
```
019-02-06 21:33:40 POSTPROCESS::RADARR: FAILED DOWNLOAD DETECTED FOR The.Wave.2015.720p.BluRay.x264-PSYCHD
2019-02-06 21:33:40 POSTPROCESS::RADARR: SUCCESS: Sending failed download to Radarr for CDH processing
2019-02-06 21:33:40 ERROR   ::MAIN: A problem was reported in the /volume1/@appstore/nzbget/share/nzbget/scripts/nzbToMedia/nzbToRadarr.py script.
```
Note the `ERROR` line is reported by NZBGet when we return the failed flag... but hopefully the `SUCCESS` line in the log helps verify that the script did complete.... It is still misleading...
